zenoh/src/api/bytes.rs

```diff
@@ -2094,12 +2222,464 @@ where
 
 impl<A, B> TryFrom<&mut ZBytes> for (A, B)
 where
-    A: TryFrom<ZBytes> + 'static,
-    <A as TryFrom<ZBytes>>::Error: Debug + 'static,
-    B: TryFrom<ZBytes> + 'static,
-    <B as TryFrom<ZBytes>>::Error: Debug + 'static,
+    A: TryFrom<ZBytes>,
+    <A as TryFrom<ZBytes>>::Error: Debug,
+    B: TryFrom<ZBytes>,
+    <B as TryFrom<ZBytes>>::Error: Debug,
 {
-    type Error = ZError;
+    type Error = ZBytesReadErrorTuple2<A, B>;
+
+    fn try_from(value: &mut ZBytes) -> Result<Self, Self::Error> {
+        ZSerde.deserialize(&*value)
+    }
+}
+
+// Tuple (a, b, c)
+macro_rules! impl_tuple3 {
```

Comment:
I guess the reason we use macro instead of inline here is type issue, right?

Reply:
Correct and trait recursiveness...